### PR TITLE
Update DEVELOPER_GUIDE.md

### PR DESCRIPTION
**Pull request by Merey1508**
### Description

New contributors to the project have been facing an issue with nox testing, which has been causing delays and difficulties in their development process. 

### Issues Resolved

To address this, an update has been made to the Developer Guide that provides more detailed instructions before running nox tests and troubleshooting common errors. Specifically, during the office hour session, there was a discussion about updating the cluster settings.
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by dhrubo-os** [comment]
Thanks for this PR, this will really be helpful. 

```
Summary

Commit sha: [ffa3052](https://github.com/opensearch-project/opensearch-py-ml/pull/107/commits/ffa3052dd91120693e3cc552daa14b25aa49407f), Author: Merey, Committer: GitHub; The sign-off is missing.
```

sign-off is missing. Can you please take care of that? Thanks.

**Comment by Merey1508** [comment]
> Thanks for this PR, this will really be helpful.
> 
> ```
> Summary
> 
> Commit sha: [ffa3052](https://github.com/opensearch-project/opensearch-py-ml/pull/107/commits/ffa3052dd91120693e3cc552daa14b25aa49407f), Author: Merey, Committer: GitHub; The sign-off is missing.
> ```
> 
> sign-off is missing. Can you please take care of that? Thanks.

Sorry, updated my commit. Is it ok now?

File: DEVELOPER_GUIDE.md [hunk 1]
```diff
@@ -77,7 +77,23 @@ OpenSearch-py-ml requires a running version of OpenSearch (from opensearch 2.5)
 You can install opensearch multiple ways:
 
 1. https://opensearch.org/downloads.html#docker-compose
-2. https://opensearch.org/docs/2.5/install-and-configure/install-opensearch/tar/
+
+After navigating to OpenSearch Dashboards you should update the persistent settings for the cluster. The settings will update the behavior of the machine learning plugin, specifically the ml_commons plugin. 
+
+You should paste this settings in the `Dev Tools` window and run it:
```
Comment (dhrubo-os) [inline DEVELOPER_GUIDE.md:83]: Can you please add this link: https://opensearch.org/docs/latest/ml-commons-plugin/cluster-settings/ as reference.

**Comment by dhrubo-os** [comment]
> > Thanks for this PR, this will really be helpful.
> > ```
> > Summary
> > 
> > Commit sha: [ffa3052](https://github.com/opensearch-project/opensearch-py-ml/pull/107/commits/ffa3052dd91120693e3cc552daa14b25aa49407f), Author: Merey, Committer: GitHub; The sign-off is missing.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > sign-off is missing. Can you please take care of that? Thanks.
> 
> Sorry, updated my commit. Is it ok now?

Still showing the same. 

https://github.com/opensearch-project/opensearch-py-ml/pull/107/checks?check_run_id=12087020270

**Comment by dhrubo-os** [comment]
> > > Thanks for this PR, this will really be helpful.
> > > ```
> > > Summary
> > > 
> > > Commit sha: [ffa3052](https://github.com/opensearch-project/opensearch-py-ml/pull/107/commits/ffa3052dd91120693e3cc552daa14b25aa49407f), Author: Merey, Committer: GitHub; The sign-off is missing.
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > sign-off is missing. Can you please take care of that? Thanks.
> > 
> > 
> > Sorry, updated my commit. Is it ok now?
> 
> Still showing the same.
> 
> https://github.com/opensearch-project/opensearch-py-ml/pull/107/checks?check_run_id=12087020270

Summary

```
Commit sha: [ffa3052](https://github.com/opensearch-project/opensearch-py-ml/pull/107/commits/ffa3052dd91120693e3cc552daa14b25aa49407f), Author: Merey, Committer: GitHub; The sign-off is missing.
Commit sha: [4b2d072](https://github.com/opensearch-project/opensearch-py-ml/pull/107/commits/4b2d072c250d2e914da467486064e2a37d1c4ff1), Author: Merey, Committer: GitHub; The sign-off is missing.
```

For sign-off I usually add `-s / --signoff in my commit command. 

Check this out: https://docs.pi-hole.net/guides/github/how-to-signoff/

**Comment by Merey1508** [comment]
> > > > Thanks for this PR, this will really be helpful.
> > > > ```
> > > > Summary
> > > > 
> > > > Commit sha: [ffa3052](https://github.com/opensearch-project/opensearch-py-ml/pull/107/commits/ffa3052dd91120693e3cc552daa14b25aa49407f), Author: Merey, Committer: GitHub; The sign-off is missing.
> > > > ```
> > > > 
> > > > 
> > > >     
> > > >       
> > > >     
> > > > 
> > > >       
> > > >     
> > > > 
> > > >     
> > > >   
> > > > sign-off is missing. Can you please take care of that? Thanks.
> > > 
> > > 
> > > Sorry, updated my commit. Is it ok now?
> > 
> > 
> > Still showing the same.
> > https://github.com/opensearch-project/opensearch-py-ml/pull/107/checks?check_run_id=12087020270
> 
> Summary
> 
> ```
> Commit sha: [ffa3052](https://github.com/opensearch-project/opensearch-py-ml/pull/107/commits/ffa3052dd91120693e3cc552daa14b25aa49407f), Author: Merey, Committer: GitHub; The sign-off is missing.
> Commit sha: [4b2d072](https://github.com/opensearch-project/opensearch-py-ml/pull/107/commits/4b2d072c250d2e914da467486064e2a37d1c4ff1), Author: Merey, Committer: GitHub; The sign-off is missing.
> ```
> 
> For sign-off I usually add `-s / --signoff in my commit command.
> 
> Check this out: https://docs.pi-hole.net/guides/github/how-to-signoff/

Thanks a lot for your help! 
I got the message "Commit sign-off was manually approved." 
Is it acceptable now?

**Comment by dhrubo-os** [comment]
> 



> 

https://github.com/opensearch-project/opensearch-py-ml/actions/runs/4450069408/jobs/7815923508?pr=107

I tried to manually approve. But workflow fails. So i would say please add signoff.

**Comment by dhrubo-os** [comment]
Can you access this link: https://github.com/opensearch-project/opensearch-py-ml/pull/107/checks ?

**Pull request closed without merging**